justifi-tech/web-component-library
# FE-63: Create payment submit wrapper component

*Add* new `PaymentForm` component which:
- Combines `BillingForm` component with the card/bank account components
- Allows card/bank account to be toggled in the UI
- Allows developers to specify which payment methods they want to show, and hides the toggle if not both card and bank account
- `submit` method can be called, which produces one of the following: client side errors, tokenize error (from `payment_methods`) endpoint, or a payment method token (from `payment_methods` endpoint)

*Add* ability to pass `iframeOrigin` to the following components:
- New `PaymentForm`
- `CardForm`
- `BankAccountForm`
- `PaymentMethodForm`

<img width="480" alt="Screen Shot 2023-04-12 at 11 33 58 AM" src="https://user-images.githubusercontent.com/3867103/232127469-8c7bd118-790e-497c-858d-5171c82cc6fa.png">


Links
-----



Developer considerations
--------------

  - If this is a MAJOR version, what are the breaking API changes? Was anything added to the API?
  - If this is a MINOR version, what feature was added? Why would consumers want this new version?
  - If this is a PATCH version, what was fixed?


Steps for Testing/QA
--------------------
-  Pull this branch and run `yarn && yarn start` from the `stencil-library` directory
- Go to http://localhost:6006/?path=/story/components-paymentform--basic
- In the storybook inputs section, provide a `clientId`
- In the storybook inputs section, add `name` to `paymentMethodData`
- Fill out the credit card and billing forms (note: Storybook has a bunch of keyboard shortcuts - avoid using those keys! We will need to disable the keyboard shortcuts in a future task)
- With the developer network tools open, click "Submit" - there should be a successful request to the `payment_methods` endpoint, which should return a payment method token
- Try again without `name` - the `payment_methods` request should return an error
